src/apps/vpn/interfaceconfig.h

```diff
@@ -10,8 +10,13 @@
 
 #include "ipaddress.h"
 
-struct InterfaceConfig {
-  int m_hopindex = 0;
+class QJsonObject;
+
+class InterfaceConfig {
+ public:
+  InterfaceConfig() {}
+
+  QString m_hopType;  // One of: "single", "entry", or "exit"
```

Comment:
It has to get converted into JSON in order to be serialized over the IPC connection, which doesn't support enums, but you are right that this would be cleaner if we used enums in the C++ side of things.